clang/lib/Lex/Preprocessor.cpp

```diff
@@ -925,6 +932,30 @@ void Preprocessor::Lex(Token &Result) {
     case tok::period:
       ModuleDeclState.handlePeriod();
       break;
+    case tok::annot_module_name: {
+      auto *Info = static_cast<ModuleNameInfo *>(Result.getAnnotationValue());
+      for (const auto &Tok : Info->getTokens()) {
+        switch (Tok.getKind()) {
+        case tok::identifier:
+          ModuleDeclState.handleIdentifier(Tok.getIdentifierInfo());
+          break;
+        case tok::period:
+          ModuleDeclState.handlePeriod();
```

Comment:
Given we're handling ModulesDeclState here, how about the handling in `Preprocessor::Lex`? I mean, could we remove that?

Reply:
Good catch! Removed.